_el/exercises.md

```diff
@@ -336,172 +204,52 @@ piece of code with the output it produces".
+
+Όπως και σε άλλες περιπτώσεις, είναι σημαντικό να διακρίνει κανείς την ικανοποίηση που νοιώθουν οι μαθητές από τα μαθησιακά αποτελέσματα. [[Magu2018](#CITE)] σε ένα μάθημα Πληροφορικής δευτέρου έτους αντικατέστησαν τα ανεπίσημα εργαστήρια προγραμματισμού από ένα εβδομαδιαίο τεστ που αξιολογήθηκε από ένα εργαλείο αυτόματης βαθμολόγησης και το οποίο είχε αρχικά αναπτυχθεί για διαγωνισμούς προγραμματισμού. Παρόλο που στους μαθητές δεν άρεσε το το αυτοματοποιημένο σύστημα, το συνολικό ποσοστό αποτυχίας για το μάθημα μειώθηκε κατά το ήμισυ, και ο αριθμός των μαθητών που βαθμολογήθηκαν με άριστα τριπλασιάστηκε. Αντίθετα, οι [[Rubi2014](#CITE)] άρχισαν επίσης να χρησιμοποιούν ένα αυτοματοποιημένο σύστημα βαθμολόγησης σχεδιασμένο για διαγωνισμούς, αλλά δεν είδε σημαντική μείωση των ποσοστών εγκατάλειψης των μαθητών τους - για άλλη μια φορά, οι μαθητές έκαναν κάποια αρνητικά σχόλια σχετικά με το εργαλείο, τα οποία οι συγγραφείς αποδίδουν στα μηνύματά ανατροφοδότησης του ίδιου του εργαλείου παρά στην αντιπάθεια του εργαλείου καθ' αυτού.
+
+Οι [[Srid2016](#CITE)] υιοθέτησαν μια διαφορετική προσέγγιση. Χρησιμοποίησαν [ασαφείς δοκιμές](#g:fuzz-testing) (δηλ., τυχαία παραγόμενες δοκιμασίες) για να ελέγξουν εαν ο κώδικας του μαθητή κάνει το ίδιο πράγμα με μια πρότυπη υλοποίηση που παρέχεται από τον εκπαιδευτή. Στην πρώτη εργασία ενός εισαγωγικού μαθήματος με 1400 μαθητες, οι ασαφείς δοκιμές μπόρεσαν να εντοπίσουν λάθη τα οποία ξέφυγαν από μια σειρά δοκιμαστικών περιπτώσεων που έγιναν χειρόγραφα για περισσότερο από 48% των μαθητών, γεγονός που δείχνει σαφώς την αξία του.
+
+Οι [[Basu2015](#CITE)] έδωσαν στους εκπαιδευόμενους μια σειρά από λύσεις δοκιμαστικών τεστ, αλλά οι μαθητές έπρεπε να ξεκλειδώσουν το καθένα απαντώντας σε ερωτήσεις σχετικά με την αναμενόμενη συμπεριφορά τους πριν τους επιτραπεί να το εφαρμόσουν στις προτεινόμενες από αυτές λύση. Για παράδειγμα, έστω ότι οι μαθητές γράφουν μια λειτουργία για να βρουν το μεγαλύτερο γειτονικό ζευγάρι αριθμών σε μια λίστα. Πριν να τους επιτραπεί η χρήση των δοκιμών που σχετίζονται με αυτό το πρόβλημα, οι μαθητές θα πρέπει πρώτα να επιλέξουν τη σωστή απάντηση στο ερώτημα "Ποιό είναι το αποτέλεσμα της κλήσης `largestPair(4, 3, -1, 5, 3, 3)`;" (Η σωστή απάντηση είναι «`(5, 3)`.) Σε ένα πανεπιστημιακό μάθημα 1300 ατόμων, η μεγάλη πλειοψηφία των μαθητών επέλεξε να επικυρώσει την κατανόησή των δοκιμαστικών τέστ με αυτόν τον τρόπο πριν να επιχειρήσουν να λύσουν τα προβλήματα, με αποτέλεσμα στη συνέχεια να κάνουν λιγότερες ερωτήσεις και να εκφράσουν λιγότερη σύγχυση σχετικά με το περιεχόμενο των προβλημάτων.
```

Comment:
σχετικά με την αναμενόμενη συμπεριφορά τους πριν τους επιτραπεί να το εφαρμόσουν στις προτεινόμενες από αυτές λύση. --> σχετικά με την αναμενόμενη συμπεριφορά του πριν να μπορέσουν να δοκιμάσουν τη λύση που προτείνανε.
λειτουργία --> συνάρτηση
[η μεγάλη πλειοψηφία των μαθητών ...]να επιχειρήσουν να λύσουν--> να επιχειρήσει να λύσει
να κάνουν...και να εκφράσουν-->να κανει...και να εκφράσει
Η' αντί για "η μεγάλη πλειοψηφία των μαθητων"--> οι περισσότεροι μαθητές και αφήνουμε τα ρήματα ως έχουν

Reply:
Τhanks @pdelias ! Fixed in a new commit.